library/src/main/java/io/ona/kujaku/views/KujakuMapView.java

```diff
@@ -945,11 +946,18 @@ private void changeImageButtonResource(ImageButton imageButton, int resourceId)
     private void checkPermissions() {
         if (getContext() instanceof Activity) {
             final Activity activity = (Activity) getContext();
-            PermissionListener dialogPermissionListener = new LocationPermissionListener(activity);
+            PermissionListener dialogLocationPermissionListener = new LocationPermissionListener(activity);
 
             Dexter.withActivity(activity)
                     .withPermission(Manifest.permission.ACCESS_FINE_LOCATION)
-                    .withListener(dialogPermissionListener)
+                    .withListener(dialogLocationPermissionListener)
+                    .check();
+
+            PermissionListener dialogStoragePermissionListener = new StoragePermissionListener(activity);
+
+            Dexter.withActivity(activity)
```

Comment:
Nice work 😄 

This does not work the first time you start the activity or resume the activity. Since there are multiple permissions being requested use this https://github.com/Karumi/Dexter#multiple-permissions. 

Dexter creates another activity that is uses to request permissions. Any code after the dexter request permissions will not be executed or is useless because dexter will have started another activity.

Reply:
Ok updated